### PR TITLE
Dns resolution binding

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMISVirtualEndpointGateway_AllowDnsResolutionBinding
--- PASS: TestAccIBMISVirtualEndpointGateway_AllowDnsResolutionBinding (127.11s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     129.230s
```

```
=== RUN   TestAccIBMISVirtualEndpointGatewayDataSource_AllowDnsResolutionBinding
--- PASS: TestAccIBMISVirtualEndpointGatewayDataSource_AllowDnsResolutionBinding (110.66s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     112.854s
```

```
=== RUN   TestAccIBMISVirtualEndpointGatewaysDataSource_AllowDnsResolutionBinding
--- PASS: TestAccIBMISVirtualEndpointGatewaysDataSource_AllowDnsResolutionBinding (104.28s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     106.417s
```

```
=== RUN   TestAccIBMIsVPCDnsResolutionBindingResourceBasic
--- PASS: TestAccIBMIsVPCDnsResolutionBindingResourceBasic (72.63s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     74.793s
```

```
=== RUN   TestAccIBMIsVPCDnsResolutionBindingsDataSourceBasic
--- PASS: TestAccIBMIsVPCDnsResolutionBindingsDataSourceBasic (69.61s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     71.873s
```

```
=== RUN   TestAccIBMIsVPCDnsResolutionBindingDataSourceBasic
--- PASS: TestAccIBMIsVPCDnsResolutionBindingDataSourceBasic (73.08s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     75.238s
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_vpc_test.go [hunk 1]
```diff
@@ -205,6 +256,17 @@ func testAccCheckIBMISVPCConfig(name string) string {
...
+}
+func testAccCheckIBMISVPCDnsConfig(name string) string {
+	return fmt.Sprintf(`
+	resource "ibm_is_vpc" "testacc_vpc" {
+		name = "%s"
+		default_network_acl_name = "dnwacln"
+		default_security_group_name = "dsgn"
+		default_routing_table_name = "drtn"
+		tags = ["Tag1", "tag2"]
+	}`, name)
```
Comment (astha-jain) [inline ibm/service/vpc/resource_ibm_is_vpc_test.go:268]: This configuration does not include DNS attributes?

File: examples/ibm-is-ng/main.tf [hunk 2]
```diff
@@ -1258,6 +1258,24 @@ data "ibm_is_share" "is_share" {
...
+	vpc_id  = ibm_is_vpc.vpc1.id
+  id      = ibm_is_vpc.vpc2.id
+}
+resource "ibm_is_vpc_dns_resolution_binding" "is_vpc_dns_resolution_binding" {
+	vpc_id = ibm_is_vpc.vpc1.id
+	name = "dns-bind-1"
+	vpc {
+		id = ibm_is_vpc.vpc2.id
+	}
+}
```
Comment (astha-jain) [inline examples/ibm-is-ng/main.tf:1278]: missing examples for VPC

File: website/docs/r/is_vpc.html.markdown [hunk 3]
```diff
@@ -53,6 +71,39 @@ Review the argument references that you can specify for your resource.
 - `default_network_acl_name` - (Optional, String) Enter the name of the default network access control list (ACL).
 - `default_security_group_name` - (Optional, String) Enter the name of the default security group.
 - `default_routing_table_name` - (Optional, String) Enter the name of the default routing table.
+- `dns` - (Optional, List) The DNS configuration for this VPC.
+  
+  Nested scheme for `dns`:
```
Comment (astha-jain) [inline website/docs/r/is_vpc.html.markdown:76]: Note - This feature is a Select Location availability invitation only feature. If used in other regions may lead to inconsistencies in state management.

File: website/docs/r/is_vpc_dns_resolution_binding.html.markdown [hunk 4]
```diff
@@ -0,0 +1,124 @@
+---
+layout: "ibm"
+page_title: "IBM : ibm_is_vpc_dns_resolution_binding"
+description: |-
+  create VPCDNSResolutionBinding
+subcategory: "Virtual Private Cloud API"
+---
+
+# ibm_is_vpc_dns_resolution_binding
```
Comment (astha-jain) [inline website/docs/r/is_vpc_dns_resolution_binding.html.markdown:9]: Note - This feature is a Select Location availability invitation only feature. If used in other regions may lead to inconsistencies in state management.

File: website/docs/r/is_virtual_endpoint_gateway.html.markdown [hunk 5]
```diff
@@ -88,6 +88,7 @@ Review the argument references that you can specify for your resource.
   **&#x2022;** For more information, about creating access tags, see [working with tags](https://cloud.ibm.com/docs/account?topic=account-tag&interface=ui#create-access-console).</br>
   **&#x2022;** You must have the access listed in the [Granting users access to tag resources](https://cloud.ibm.com/docs/account?topic=account-access) for `access_tags`</br>
   **&#x2022;** `access_tags` must be in the format `key:value`.
+- `allow_dns_resolution_binding` - (Optional, bool) Indicates whether to allow this endpoint gateway to participate in DNS resolution bindings with a VPC that has dns.enable_hub set to true.
```
Comment (astha-jain) [inline website/docs/r/is_virtual_endpoint_gateway.html.markdown:91]: Note - This feature is a Select Location availability invitation only feature. If used in other regions may lead to inconsistencies in state management.

Comment (ujjwal-ibm) [reply website/docs/r/is_virtual_endpoint_gateway.html.markdown:91]: image

File: website/docs/r/is_vpc_dns_resolution_binding.html.markdown (hunk 4 above)
Comment (ujjwal-ibm) [reply website/docs/r/is_vpc_dns_resolution_binding.html.markdown:9]: image

File: website/docs/r/is_vpc.html.markdown [hunk 6]
```diff
@@ -29,6 +29,41 @@ resource "ibm_is_vpc" "example" {
...
+  dns {
+		enable_hub = true
+		resolver {
+			manual_servers {
+				address = "192.168.3.4"
+			}
+		}
+	}
+}
+
```
Comment (astha-jain) [inline website/docs/r/is_vpc.html.markdown:50]: Can we add an example with zone affinity please all three zones?

**Comment by ujjwal-ibm** [comment]
moving to upstream

**Pull request closed without merging**